src/renderer/vt/VtSequences.cpp

```diff
@@ -456,3 +456,25 @@ using namespace Microsoft::Console::Render;
 {
     return _Write("\x1b[?9001h");
 }
+
+// Method Description:
+// - Formats and writes a sequence to add a hyperlink to the terminal buffer
+// Arguments:
+// - The hyperlink URI
+// Return Value:
+// - S_OK if we succeeded, else an appropriate HRESULT for failing to allocate or write.
+[[nodiscard]] HRESULT VtEngine::_SetHyperlink(std::wstring uri, USHORT id) noexcept

```

Comment:
Should we maybe split these up into `StartHyperlink(wstring, unsigned short)` and `EndHyperlink`, rather then have one overloaded method for both?

Reply:
If we split it into 2 methods then the check for which one to send would need to happen in `ActionOscDispatch` in `OutputStateMachineEngine.cpp`. I felt like that was a strange place to check for it since it is still the same OSC sequence and so it should be on the dispatcher to handle the possible inputs. Thoughts?

Reply:
So, this one is on the VT _renderer_, not the VT _parser_, but I expect that the same truth holds for both of them.

About the parser: So, this is the OutputStateMachineEngine's _job_ -- taking VT sequences that have made it all the way through the parser and assigning deeper meaning to them. I'm not Mike, but it does seem right to split them up based on the contents of the OSC.

About the renderer: probably cleaner to split them up, honestly. Little more boilerplate but it's just in this one file :smile:

Reply:
Its now 2 methods!